### PR TITLE
Provider cleanup

### PR DESCRIPTION
**Pull request by julien51**
# Description

This adds the missing part for the checkout components

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/content/CheckoutContent.tsx [hunk 1]
```diff
@@ -43,6 +44,11 @@ export const CheckoutContent = ({
   configFromSearch,
   errors,
 }: CheckoutContentProps) => {
+  const { loading } = useProvider()
+  if (loading) {
+    return <></>
```
Comment (cnasc) [inline unlock-app/src/components/content/CheckoutContent.tsx:49]: Do we want to return nothing here? That will likely cause an increase in perceived latency for the end user

Comment (julien51) [reply unlock-app/src/components/content/CheckoutContent.tsx:49]: That's a good point. I think the _pick provider_ logic will soon be moved to another place. I will address this in there.

**Comment by cnasc** [comment]
Do you think we could split this up a bit? Maybe extend the hook first, then follow up with introducing the new capabilities?

**Comment by julien51** [comment]
That's a good idea. I'll do that.

**Pull request closed without merging**